### PR TITLE
Refactoring code to make it more usable, readable and understandable

### PR DESCRIPTION
**Pull request by betobetodev**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,295 +1,43 @@
...
-    - [List applications status](#list-applications-status)
-- [Plugin creation](#plugin-creation)
-- [License](#license)
+## Overview
+The Broker component is the framework entry point for the user. It is responsible for receiving a submission from the user and preparing its execution.
 
-# Architecture description
---------
-The Manager is implemented following a plugin architecture, providing flexibility to add or remove plugins when necessary. All the integrations with different infrastructures and components are made by specific plugins, so the different technologies in the context of EUBra-BIGSEA framework can be easily integrated by the broker service.
+To more info, see [details.md](https://github.com/bigsea-ufcg/bigsea-manager/tree/refactor/docs/details.md)
```
Comment (igornatanael) [inline README.md]: the path for the details.md must point to the master branch. so you have to use https://github.com/bigsea-ufcg/bigsea-manager/tree/master/docs/details.md

File: README.md [hunk 2]
```diff
@@ -1,295 +1,43 @@
...
 
-# Dependencies
---------
-To have your Manager working properly you need to ensure that it has access to following components in your infrastructure:
+## How does it works?
+The broker is implemented following a **plugin architecture**, providing flexibility to add or remove plugins when necessary.
```
Comment (igornatanael) [inline README.md]: suggestion: change "providing flexibility to add or remove plugins when necessary" to "providing flexibility to customize your deployment using only the plugins you need, avoiding to include unnecessary dependencies (from others plugins) to your deploy environment."

File: README.md [hunk 3]
```diff
@@ -1,295 +1,43 @@
...
+All the integrations with different infrastructures and components are made by specific plugins.
 
-* **OpenStack Compute Service** - *Nova* (with admin privileges)
-* **OpenStack Monitoring Service** - *Monasca*
+## How to develop a plugin?
+See [plugin-development.md](https://github.com/bigsea-ufcg/bigsea-manager/tree/refactor/docs/plugin-development.md).
```
Comment (igornatanael) [inline README.md]: the same comment about the branch in the file path

File: README.md [hunk 4]
```diff
@@ -1,295 +1,43 @@
...
 
-# Deploy
-To install, configure and run the Manager component you will need a virtual machine with the configurations described below.
+## Requirements
+* Python 2.7 or Python 3.5
```
Comment (igornatanael) [inline README.md]: does it really work well with python 3?
I don't think so

File: README.md [hunk 5]
```diff
@@ -1,295 +1,43 @@
...
+* Linux packages: python-dev and python-pip
```
Comment (igornatanael) [inline README.md]: I think that there are others, as ssl and some packages that you have to install to have python-dev working properly.

File: README.md [hunk 6]
```diff
@@ -1,295 +1,43 @@
...
-**Minimal configuration**
-```
-OS: Ubuntu 14.04
-CPU: 1 core
-Memory: 2GB of RAM
-Disk: there is no disk requirements.
-```
-
-In the virtual machine that you want to install the manager follow the steps below:
+To **apt** distros, you can use [pre-install.sh](https://github.com/bigsea-ufcg/bigsea-manager/tree/refactor/tools/pre-install.sh) to install the requirements.
```
Comment (igornatanael) [inline README.md]: the same comment about the branch in the file path

File: README.md [hunk 7]
```diff
@@ -1,295 +1,43 @@
...
-| metric_source | string |  |
-| application_type | string |  |
-| check_interval | int |  |
-| trigger_down | int |  |
-| trigger_up | int |  |
-| min_cap | int |  |
-| max_cap | int |  |
-| actuation_size | int |  |
-| metric_rounding | int |  |
+First of all, install **git**. After, you just need to clone the [Broker repository](https://github.com/bigsea-ufcg/bigsea-manager.git) in your machine.
```
Comment (igornatanael) [inline README.md]: I don't this it is necessary this step about install git. Too much dump.

File: README.md [hunk 8]
```diff
@@ -1,295 +1,43 @@
...
-}
-```
-
-#### Response parameters
-| Name | Type | Description |
-| --- | --- | --- |
-| execution_id | string |  |
+### Configuration
+A configuration file is required to run the Broker. **Edit and fill your broker.cfg in the root of Broker directory.** Make sure you have fill up all fields before run.
+You can find a template in [config-example.md](https://github.com/bigsea-ufcg/bigsea-manager/tree/refactor/docs/config-example.md).
```
Comment (igornatanael) [inline README.md]: same comment about the branch in the file path

File: README.md [hunk 9]
```diff
@@ -1,295 +1,43 @@
...
+You can find a template in [config-example.md](https://github.com/bigsea-ufcg/bigsea-manager/tree/refactor/docs/config-example.md). 
 
-#### Response example
+### Run
+In the Broker root directory, start the service using tox command:
 ```
-{  
-   "execution_id":"osspark04"
-}
+$ tox -e venv -- broker
```
Comment (igornatanael) [inline README.md]: the pattern decided is that we will always use ./run.sh, even if this file contains only the tox commands. 
it is a strategy to keep all the components easy to run and install and with the same command for all of they.

File: README.md [hunk 10]
```diff
@@ -1,295 +1,43 @@
...
-- **execute(self, data)**
-  - Actually execute the logic of cluster creation and job execution
-  - Returns information if the execution was successful or not
-    
-#### Example:
-
-```
-from application_manager.plugins import base
+## Broker REST API
+Endpoints is avaliable on [restapi-endpoints.md](https://github.com/bigsea-ufcg/bigsea-manager/tree/refactor/docs/restapi-endpoints.md) documentation.
```
Comment (igornatanael) [inline README.md]: "Endpoints is" > "Endpoints are"

Comment (igornatanael) [inline README.md]: same comment about the branch in the file path

File: broker/utils/framework/monitor.py [hunk 11]
```diff
@@ -21,7 +21,7 @@
 def _get_monitor_data(plugin, info_plugin, collect_period):
     start_monitor_dict = {
         'plugin': plugin,
-        'info_plugin': info_plugin,
+        'plugin_info': info_plugin,
```
Comment (igornatanael) [inline broker/utils/framework/monitor.py]: also change the value "info_plugin" to "plugin_info"

File: docs/asperathos-workflow.md [hunk 12]
```diff
@@ -0,0 +1,11 @@
+# Asperathos execution workflow
+![Flow](https://github.com/bigsea-ufcg/bigsea-manager/blob/refactor/docs/flow.png)
```
Comment (igornatanael) [inline docs/asperathos-workflow.md]: same comment about the branch in the file path

File: docs/config-example.md [hunk 13]
```diff
@@ -1,13 +1,21 @@
+# Config example
```
Comment (igornatanael) [inline docs/config-example.md:1]: maybe would be better to move the plugins specific information to each plugin description. 
imagine the day we have 100 plugins, so this example will have 100 fields? you can add only two plugins fields here called "plugin-#" and make like this:

`[plugin-1]
p1_info1 = lorem
p1_info2 = ipsum

[plugin-2]
p2_info1 = lorem_ipsum
p2_info2 = ipsum_lorem
`

File: docs/installation.md [hunk 14]
```diff
@@ -0,0 +1,19 @@
+## Requirements
+* Python 2.7 or Python 3.5
+* Linux packages: python-dev and python-pip
+* Python packages: setuptools, tox and flake8
+
+To **apt** distros, you can use [pre-install.sh](https://github.com/bigsea-ufcg/bigsea-manager/tree/refactor/tools/pre-install.sh) to install the requirements.
+
+## Install
+First of all, install **git**. After, you just need to clone the [Broker repository](https://github.com/bigsea-ufcg/bigsea-manager.git) in your machine.
```
Comment (igornatanael) [inline docs/installation.md]: unnecessary git install step (again)

File: docs/installation.md [hunk 15]
```diff
@@ -0,0 +1,19 @@
+## Requirements
+* Python 2.7 or Python 3.5
+* Linux packages: python-dev and python-pip
+* Python packages: setuptools, tox and flake8
+
+To **apt** distros, you can use [pre-install.sh](https://github.com/bigsea-ufcg/bigsea-manager/tree/refactor/tools/pre-install.sh) to install the requirements.
```
Comment (igornatanael) [inline docs/installation.md]: again, change branch on the file path link

File: tools/pre-install.sh [hunk 16]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
```
Comment (igornatanael) [inline tools/pre-install.sh]: move this script to root

File: tools/run.sh [hunk 17]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
```
Comment (igornatanael) [inline tools/run.sh]: move this script to root

**Review by igornatanael — CHANGES_REQUESTED** [review]
some suggestions and change requests

**Comment by betobetodev** [comment]
@igornatanael check the changes, please.

**Review by igornatanael — APPROVED** [review]

**Pull request merged**